platform/mk2/hal/timer_stm32/timer_device_stm32.c

```diff
@@ -340,11 +341,17 @@ bool timer_device_init(const size_t chan, const uint32_t speed,
         return false;
 }
 
-void timer_device_reset_count(size_t chan) {}
+void timer_device_reset_count(size_t chan) 
```

Comment:
make this const and normally we prefer something like `if (chan >= MK2_TIMER_CHANNELS) return;` as this avoids indentation and is easier to expand upon.